test/apps/iodemo/io_demo.cc

```diff
@@ -876,13 +876,13 @@ class DemoServer : public P2pDemoCommon {
             }
 
             if (_status == UCS_OK) {
-                ASSERTV(_conn->ucx_status() == UCS_OK) << "conn_status="
-                        << _conn->ucx_status();
-
-                _server->send_io_write_response(_conn, *_iov, _sn);
                 if (_server->opts().validate) {
                     validate(*_iov, _sn);
                 }
+                
+                if (_conn->ucx_status() == UCS_OK) {
```

Comment:
should we validate only if _conn->ucx_status() == UCS_OK?

Reply:
according to our offline discussion, we shouldn't check the connection's status, since if some operation is completed with OK, the data should be valid